kuka_rox_hardware_interface/src/rox_hardware_interface.cpp

```diff
@@ -149,13 +162,21 @@ CallbackReturn KukaRoXHardwareInterface::on_activate(const rclcpp_lifecycle::Sta
   request.set_external_control_mode(
     kuka::motion::external::ExternalControlMode(
       std::stoi(info_.hardware_parameters.at("control_mode"))));
+  RCLCPP_INFO(
+    rclcpp::get_logger("KukaRoXHardwareInterface"), "Starting control in %s",
+    kuka::motion::external::ExternalControlMode_Name(
+      std::stoi(
+        info_.hardware_parameters.at(
+          "control_mode"))).c_str());
 
   if (stub_->OpenControlChannel(
       &context, request,
       &response)
     .error_code() != grpc::StatusCode::OK)
   {
-    return CallbackReturn::ERROR;
+    RCLCPP_ERROR(rclcpp::get_logger("KukaRoXHardwareInterface"), "OpenControlChannel failed");
+    terminate_ = true;
```

Comment:
Is it possible to only spawn the thread if the OpenControlChannel grpc call is successful, at the end of the on_activate function? It might be possible to avoid setting the terminate_ flag here that way and maybe find a less hacky way to restart it if the activation fails.

Reply:
The ObserControl request must be called before the OpenControlChannel, as otherwise we would miss the first two state changes (in which case the RT loop would not start)